src/subscription_manager/injectioninit.py

```diff
@@ -50,6 +50,10 @@ def init_dep_injection():
 
     This needs to be called from any entry-point into subscription manager.
     """
+    # Enable or disable status messages. Note that this reflects only the CLI
+    # input, rhsm.conf may be set differently.
+    inj.provide(inj.PROGRESS_MESSAGES, True)
```

Comment:
Why use this in addition to the value of the new configuration parameter?

Reply:
The problem with the CLI option is that we can't properly distribute it through the code without making major changes. Some other CLI options (proxy settings etc.) are set directly in the CLI code (`self.cp_provider.set_connection_info()`). But we can't do that here, because we'd have to distribute it all the way to all the individual functions in `rhsmlib/connection.py` as arguments.

I'm not 100% content with this solution, but I didn't think of anything else that would have lower impact.

Reply:
Of course, the better solution to this would be to make a singleton, but we are not using them in our code base (yet). When we get to the moment when we're using singletons and not injections, this will look cleaner, but I wanted to be consistent.

Reply:
Sounds great to me! Thank you for the response!